### PR TITLE
Add better NFC status callbacks

### PR DESCRIPTION
**Pull request by Franzaine**
Solves the issues presented in #26

### CODE REVIEW

**Comment by Franzaine** [comment]
@petter-fidesmo I'm having a hard time building this, could you assist?

File: android/src/main/java/nordpol/TagDispatcher.java [hunk 1]
```diff
@@ -79,9 +87,10 @@ public boolean disableExclusiveNfc() {
             } else {
                 disableForegroundDispatch(adapter);
             }
-            return true;
+            return NfcStatus.AVAILABLE;
         }
-        return false;
+        if (handleUnavailableNfc) toast("NFC is not available on this device");
```
Comment (petter-fidesmo) [inline android/src/main/java/nordpol/TagDispatcher.java]: @Franzaine I think this is also maybe a bit overkill?

File: android/src/main/java/nordpol/TagDispatcher.java [hunk 2]
```diff
@@ -10,68 +10,76 @@
...
 
 public class TagDispatcher {
     private static final int DELAY_PRESENCE = 5000;
     
     private OnDiscoveredTagListener tagDiscoveredListener;
-    private OnNfcDisabledListener nfcDisabledListener;
+    private boolean handleUnavailableNfc;
     private Activity activity;
 
+    public enum NfcStatus {
```
Comment (petter-fidesmo) [inline android/src/main/java/nordpol/TagDispatcher.java]: @Franzaine If we do not return status for disabled, then the AVAILABLE state is not required and I think we should remove it.

File: android/src/main/java/nordpol/TagDispatcher.java [hunk 3]
```diff
@@ -10,68 +10,76 @@
...
 
     /** Disable exclusive NFC access for the given activity.
-     * @returns true if NFC was available and false if no NFC is available
-     *          on device.
+     * @returns NfcStatus.AVAILABLE if NFC was available and
+     * NfcStatus.NOT_AVAILABLE if no NFC is available on the device.
      */
     @TargetApi(Build.VERSION_CODES.KITKAT)
-    public boolean disableExclusiveNfc() {
+    public NfcStatus disableExclusiveNfc() {
```
Comment (petter-fidesmo) [inline android/src/main/java/nordpol/TagDispatcher.java]: @Franzaine Maybe this doesn't really makes sense. I guess we could just as well make this function void?

Comment (Franzaine) [inline android/src/main/java/nordpol/TagDispatcher.java]: Agree!

File: android/src/main/java/nordpol/TagDispatcher.java (hunk 1 above)
Comment (Franzaine) [inline android/src/main/java/nordpol/TagDispatcher.java]: I'd argue that it is not overkill. The parameter says that the library should handle unavailable NFC. If we don't do this the implementer still has to get the adapter and all to even have basic support. This way we facilitate a pretty good UX right from the get-go.

File: android/src/main/java/nordpol/TagDispatcher.java (hunk 2 above)
Comment (Franzaine) [inline android/src/main/java/nordpol/TagDispatcher.java]: done

**Comment by Franzaine** [comment]
Ready for review @petter-fidesmo

File: android/src/main/java/nordpol/TagDispatcher.java (hunk 1 above)
Comment (petter-fidesmo) [inline android/src/main/java/nordpol/TagDispatcher.java]: @Franzaine The function promises to disable NFC. If there is no NFC the promise has still been kept, so there is no need to handle unavailable NFC. No action is needed by the user so no information is required,

**Comment by petter-fidesmo** [comment]
@Franzaine :+1: This looks very nice.

@adriaan-telcred Do you have any comments on the final result here?

**Comment by Franzaine** [comment]
Great! I'll move ahead with merging this then!

**Pull request merged**